utils/staking-miner/src/main.rs

```diff
@@ -317,40 +317,41 @@ struct MonitorConfig {
 
 	/// The solver algorithm to use.
 	#[clap(subcommand)]
-	solver: Solvers,
+	solver: Solver,
 }
 
-#[derive(Debug, Clone, Parser)]
+#[derive(Debug, Clone, Parser, PartialEq)]
 struct EmergencySolutionConfig {
 	/// The block hash at which scraping happens. If none is provided, the latest head is used.
 	#[clap(long)]
 	at: Option<Hash>,
 
 	/// The solver algorithm to use.
 	#[clap(subcommand)]
-	solver: Solvers,
+	solver: Solver,
 
 	/// The number of top backed winners to take. All are taken, if not provided.
 	take: Option<usize>,
 }
 
-#[derive(Debug, Clone, Parser)]
+#[derive(Debug, Clone, Parser, PartialEq)]
 struct DryRunConfig {
 	/// The block hash at which scraping happens. If none is provided, the latest head is used.
 	#[clap(long)]
 	at: Option<Hash>,
 
 	/// The solver algorithm to use.
 	#[clap(subcommand)]
-	solver: Solvers,
+	solver: Solver,
 
 	/// Force create a new snapshot, else expect one to exist onchain.
 	#[clap(long)]
 	force_snapshot: bool,
 }
 
-#[derive(Debug, Clone, Parser)]
-struct SharedConfig {
+#[derive(Debug, Clone, Parser, PartialEq)]
+#[clap(author, version, about)]
```

Comment:
https://github.com/clap-rs/clap/blob/v3.1.5/examples/derive_ref/README.md#command-attributes

without `--version` here the version field is elided AFAIU